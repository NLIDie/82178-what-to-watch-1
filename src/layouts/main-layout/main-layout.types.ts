export type TProps = {
  header: React.ReactNode;
  footer: React.ReactNode;
  children: React.ReactNode;
};
