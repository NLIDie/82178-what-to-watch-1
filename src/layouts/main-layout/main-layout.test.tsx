import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {MainLayout} from "./main-layout";

describe(`Layouts/main-layout`, () => {
  it(`Correct renderer layout`, () => {
    function Header(): JSX.Element {
      return <header>Header</header>;
    }

    function Footer(): JSX.Element {
      return <footer>Footer</footer>;
    }

    const tree = create(
        <MainLayout header={<Header />} footer={<Footer />}>
          <div>Main Content</div>
        </MainLayout>
    );

    expect(tree).toMatchSnapshot();
  });
});
