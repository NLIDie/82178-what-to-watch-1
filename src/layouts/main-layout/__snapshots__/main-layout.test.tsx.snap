// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts/main-layout Correct renderer layout 1`] = `
Array [
  <header>
    Header
  </header>,
  <main
    className="page-content"
  >
    <div>
      Main Content
    </div>
    <footer>
      Footer
    </footer>
  </main>,
]
`;
